pubtools/pulplib/_impl/client/retry.py

```diff
@@ -27,7 +29,16 @@ def should_retry(self, attempt, future):
         # at the error in more detail in order to decide whether to retry.
         retry = self._delegate.should_retry(attempt, future)
 
-        if future.exception() and retry:
+        exception = future.exception()
+        if exception and hasattr(exception, "response"):
+            # if request is querying maintenance repo, it's possible there's
+            # no report existed, not wasting time on retry.
+            url = str(exception.response.url)
+            status_code = exception.response.status_code
+            if url.endswith(MAINTENANCE_REPROT_URL) and status_code == 404:
```

Comment:
```suggestion
            if url.endswith(MAINTENANCE_REPORT_URL) and status_code == 404:
```

Similar as the comment elsewhere, do we actually need to connect this with maintenance mode?  Could we just never retry on 404 (at least, until we find some reason it needs a retry)?